### PR TITLE
Webpacker react

### PR DESCRIPTION
**Pull request by Code47X**
Initial webpacker setup so that we can use react components in rails views.

### CODE REVIEW

**Comment by Code47X** [comment]
@rogerroelofs @aterrype Everything seems to be good to go now, let me know if you have any thoughts. The new chart components are fully functional but could still go for a bit of styling. I figured it would be good to get this before we run into too many conflicts from continued work.

**Comment by aterrype** [comment]
This looks good from a functionality standpoint, but I think we need at least a little more styling before merging it into main. Right now it's hard to tell the difference between the lines on the graphs because they're all the same color, which seems like enough of a functional issue that I wouldn't want to put it in front of an end user.

**Comment by Code47X** [comment]
Was thinking the same thing, wasn't sure if this ci pipeline automatically deployed on merge to main. I wouldn't want it to be deployed with that styling. I've already made it so everyone has unique colors, I'll push up when I finish up the rest of the styling/interactions.

**Comment by aterrype** [comment]
I don't think it automatically deploys, though I also know that Peter is actively trying to get the deploy pipeline working again, so merging something we don't want deployed yet might throw a wrench in that.

**Comment by Code47X** [comment]
Cleaned up styling + added a few minor onHover effects.

image

File: app/models/salary_graph.rb
Comment (rogerroelofs) [inline app/models/salary_graph.rb:11]: We could simplify this by just passing the date and making the hash in employees_row_for instead of here.

File: app/models/salary_graph.rb
Comment (rogerroelofs) [inline app/models/salary_graph.rb:17]: Sorry, that was a bad name I picked.  How about something like `chart_data = { date: date } ` and inside the each `chart_data[key] = { name: e.display_name, salary: e.salary_on(date) }` ?

**Comment by rogerroelofs** [comment]
You could remove experience_graph.rb and it's related spec since it's no longer used.

**Comment by Code47X** [comment]
I decided keep the experience_graph model so that it is in line with the salary_graph. These models will handle formatting the data before passing it into the components.

**Comment by Code47X** [comment]
https://user-images.githubusercontent.com/5169367/111846501-fa606e00-88d4-11eb-93e4-3ef7ca0f88d5.mp4

File: .travis.yml
Comment (rogerroelofs) [inline .travis.yml:5]: this whole file is going away.  You may want to coordinate with @aterrype about setting up this step in the github CI action she's working on.

Comment (aterrype) [reply .travis.yml:5]: I'm planning to merge my PR in today, since it's approved now. I think it makes more sense to add node to the new CI steps in this PR rather than mine, because without the rest of this PR we aren't doing anything with node so it's harder to be sure that the setup is working.

Comment (aterrype) [reply .travis.yml:5]: Just merged my PR (and updated the project CI rule).

Comment (Code47X) [reply .travis.yml]: Added node/yarn to the CI pipeline. Looks like its working correctly.

**Review by rogerroelofs — APPROVED** [review]

**Review by aterrype — APPROVED** [review]

**Pull request merged**
